terraform/azure/main.tf

```diff
@@ -0,0 +1,174 @@
+  tags                = var.tags
+}
+
+resource "azurerm_role_assignment" "acr" {
+  scope                = azurerm_container_registry.registry.id
+  role_definition_name = "AcrPull"
+  principal_id         = azurerm_user_assigned_identity.identity.principal_id
+}
+
+resource "azurerm_key_vault" "kv" {
+  name                = var.key_vault_name
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
+  tags                = var.tags
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azuread_client_config.current.object_id
+
+    key_permissions = [
+      "Get", "Sign"
+    ]
+
+    secret_permissions = [
+      "Get", "Set", "List", "Delete", "Purge"
+    ]
+
+    certificate_permissions = [
+      "Get", "Create", "Delete", "Purge"
+    ]
+  }
+}
+
+resource "azurerm_key_vault_access_policy" "workload-identity" {
+  key_vault_id = azurerm_key_vault.kv.id
+  tenant_id    = data.azurerm_client_config.current.tenant_id
+  object_id    = azurerm_user_assigned_identity.identity.principal_id
+
+  certificate_permissions = [
+    "Get"
+  ]
+}
+
+
+resource "azurerm_kubernetes_cluster" "aks" {
+  name                      = var.cluster_name
+  location                  = azurerm_resource_group.rg.location
+  resource_group_name       = azurerm_resource_group.rg.name
+  dns_prefix                = "${var.cluster_name}-dns"
+  kubernetes_version        = "1.26.3"
+  workload_identity_enabled = true
+  oidc_issuer_enabled       = true
+
+  default_node_pool {
+    name       = "default"
+    node_count = 3
+    vm_size    = "Standard_D2_v2"
+  }
+
+  web_app_routing {
+    dns_zone_id = ""
+  }
+
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  tags = var.tags
+}
+
+resource "azurerm_role_assignment" "linkAcrAks" {
+  principal_id                     = azurerm_kubernetes_cluster.aks.kubelet_identity[0].object_id
+  role_definition_name             = "AcrPull"
+  scope                            = azurerm_container_registry.registry.id
+  skip_service_principal_aad_check = true
+}
+
+
+resource "azurerm_federated_identity_credential" "workload-identity-credential" {
+  name                = "ratify-federated-credential"
+  resource_group_name = azurerm_resource_group.rg.name
+  audience            = ["api://AzureADTokenExchange"]
+  issuer              = azurerm_kubernetes_cluster.aks.oidc_issuer_url
+  parent_id           = azurerm_user_assigned_identity.identity.id
+  subject             = "system:serviceaccount:${var.ratify_namespace}:ratify-admin"
+}
+
+resource "azurerm_key_vault_certificate" "ratify-cert" {
+  name         = var.ratify_cert_name
+  key_vault_id = azurerm_key_vault.kv.id
+
+  certificate_policy {
+    issuer_parameters {
+      name = "Self"
+    }
+
+    key_properties {
+      exportable = true
```

Comment:
why does the private key need to be exportable?

Reply:
same questions as Akash, everything else looks good, thanks!

Reply:
For Ratify, no I don't believe so. Both of you would be able to answer that better than I can, but for Notation, maybe? Either way it doesn't need to be exported for the sake of this config. I'll set that to false. :)